pkg/api/store/workload/workload_store.go

```diff
@@ -327,3 +329,22 @@ func getDomain(image string) string {
 	}
 	return domain
 }
+
+func imageUpdated(apiContext *types.APIContext, data map[string]interface{}) (bool, error) {
+	var workload projectclient.Workload
```

Comment:
if we are planning to fetch existing container from the API, we should limit it to the case when there are private registries setup. Something in the line of:

* fetch image pull secrets based on data
* if not empty, do imageUpdated call.
* set image pull secrets on data if imageUpdated is true.

Reply:
thank you, updated